core/arch/arm/kernel/boot.c

```diff
@@ -118,30 +118,41 @@ __weak unsigned long plat_get_aslr_seed(void)
 	return 0;
 }
 
-#if defined(_CFG_CORE_STACK_PROTECTOR)
+#if defined(_CFG_CORE_STACK_PROTECTOR) || defined(CFG_WITH_STACK_CANARIES)
 /* Generate random stack canary value on boot up */
-__weak uintptr_t plat_get_random_stack_canary(void)
+__weak void __nostackcheck plat_get_random_stack_canaries(void *buf, size_t blen)
 {
-	uintptr_t canary = 0xbaaaad00;
+	assert(buf && blen >= sizeof(uintptr_t) && !(blen % sizeof(uintptr_t)));
+
 	TEE_Result ret = TEE_ERROR_GENERIC;
+	void *canaries_buf = malloc(blen);
+
+	if (!canaries_buf)
+		panic("Failed to malloc canaries tmp buffer");
+
+	memset(canaries_buf, 0xab, blen);
```

Comment:
Why?

Reply:
The platform with `CFG_NS_VIRTUALIZATION` need to fill fixed value to initialize cancaries' buffer. I'll move this inside `IS_ENABLED(CFG_VIRTUALIZATION)` case.